scripts/generators/tsconfig.js

```diff
@@ -0,0 +1,66 @@
+"use strict";
+
+const path = require("path");
+const fs = require("fs");
+
+const convertedPackages = [
+  // Add .ts packages here
+  "template",
+  //"types",
+];
+
+const root = path.resolve(__dirname, "../../");
+
+const tsPkgs = convertedPackages.map(name => ({
+  name: `@babel/${name}`,
+  dir: path.resolve(root, "packages", `babel-${name}`),
+}));
+
+for (const { dir } of tsPkgs) {
+  const pkg = require(`${dir}/package.json`);
+
+  const references = [];
+  for (const dep of Object.keys(pkg.dependencies)) {
+    if (!dep.startsWith("@babel/")) continue;
+    for (const { name, dir: depDir } of tsPkgs) {
+      if (name === dep) {
+        references.push({ path: path.relative(dir, depDir) });
+        break;
+      }
+    }
+  }
+
+  fs.writeFileSync(
+    path.resolve(dir, "tsconfig.json"),
+    JSON.stringify(
+      {
+        extends: "../../tsconfig.base.json",
+        compilerOptions: {
+          // Until we have converted every package, we cannot store
+          // .d.ts files inside lib/ because it causes conflicts
+          // with Babel-related type definitions in node_modules/@types
+          outDir: "./dts",
```

Comment:
This is because, until all of our packages have `.ts` files, we:
1) Rely on some packages that are still `.js` filed and have the corresponding `node_moduels/@types/` definitions
2) Those packages rely on our already converted packages
3) They load the `.d.ts` files (_only if we store them in `lib`_)
4) TS refuses to update the `.d.ts` files because they are used as inputs

Reply:
This is a tricky one, the issue is happening only when having kinda circular dependency - in other words, generated `d.ts` files in a package are somehow imported by itself while compiling. And this is something that needs to be fixed, while migrating.

I would suggest, `@babel/template` is just not the right starting point - dependencies of it can be migrated first, and that would solve the issue.